2016/proposals/filip-ter.md

```diff
@@ -0,0 +1,312 @@
+	- `pandas-sparse` will have its own test of course, but the `sparse` module already has it's own tests so this should not be too difficult to refactor
+	
+* Implement enhancements:
+ 
+  - List of enhancements to implement: [here](https://github.com/pydata/pandas/issues?utf8=✓&q=is%3Aopen+is%3Aissue+label%3ASparse+label%3AEnhancement), if there is not enough time to for them all, I will choose a subset to implement. 
+  - This section will involve writing tests for the enhancements that will be implemented.
+
+
+--- 
+
+###Technical Details
+
+- Goal 1:
+	
+	Fixing the bugs will involve modifying existing code, making existing test pass, and perhaps adding tests that didn't catch the bug. As usual when working on Pandas issues I will add a reference to the Github issue that it relates to in the code for the fix.
+	
+	For example if I were to implement a fix for bug [#9850](https://github.com/pydata/pandas/issues/9850) in `sparse/series.py`
+	
+	```python
+		...
+		
+		def to_frame():
+			#GH9850
+		
+			#... fix bug here
+			return SparseDataFrame(...)
+		...
+	
+	```
+	
+	
+
+- Goal 2:
+		
+	A new repository `pandas-sparse` will be created, hosted on Github under `pydata`. 
+	
+	Example Usage:
+	
+	```python
+		from pandas_sparse import SparseDataFrame
+		
+		df = SparseDataFrame({'A': [1,0,2,0,0], 'B': [0,0,0,7,0],}, fill_value=0)
+		
+	```
+	The fill_value is equal to `np.nan` by default 
+	
+	
+	As mentioned before `pandas-sparse` will be an external dependency of pandas, therefore I will implement conversion methods to pandas that will allow dense data structures to be converted to sparse and vice versa. These methods currently work on the 
+	
+	Example Usage with `Series` will be the same for other data structures, this could also be an example of a test of the conversion methods:
+	
+	```python
+		import numpy as np
+		import padnas as pd
+		
+		series = pd.Series(np.random.randn(5), index=['a', 'b', 'c', 'd', 'e'])
+		
+		##Will return instance of pandas_sparse.SparseSeries
+		sparse_series = series.to_sparse()
+		
+		##The from_sparse method will convert from sparse data structures back to dense data structures
+		original_series = Series.from_sparse(sparse_series)
+		
+		##This test should pass if the methods are implemented correctly
+		assert_series_equal(series, original_series)
+		
+	```
+	
+	Snippet of what the implementation of the `to_sparse` methods in `pandas/core/series.py` will look like:
+	It is virtually the same as before except the import will come from an external project.
+	
+	```python
+		...
+		
+		def to_sparse():
+			...			
+			
+			from pandas_sparse import SparseSeries
+        	return SparseSeries(self, kind=self.kind,
+                	            	   fill_value=self.fill_value)
+        
+        ...
+    ```
+   	
+	Similarly the `SparseSeries` in `pandas-sparse` will have a `to_dense` method
+	
+	
+	 
+    ```python		
+		...
+		
+		def to_dense():
+			...
```

Comment:
@jreback Perhaps I could instead add DataFrame.from_sparse() to the core? I guess this would be a problem as it would require an import of pandas from the `pandas-sparse` files